lumicks/pylake/file.py

```diff
@@ -39,7 +39,7 @@ def __init__(self, filename):
     def from_h5py(cls, h5py_file):
         """Directly load an existing `h5py.File`"""
         new_file = cls.__new__(cls)
-        super(cls, new_file).__init__(h5py_file)
+        super(File, new_file).__init__(h5py_file)
```

Comment:
I'm not really following the explanation for this change in the commit message. The first argument to `super()` should be the derived class, not the base.

Reply:
I'd then like to invite you to actually try and derive a child class, and have it call into `from_h5py`. It won't work, as it'll end up calling `File.__init__`, which doesn't take an `h5` file as its argument.

Reply:
Ah, indeed, sorry. I seriously made a mess of the structure here because it's actually expecting the `Group.__init__` there but even that's dubious because of the multiple mixin classes. I'd suggest removing `super()` altogether and just going for a simple and explicit `new_file.h5 = h5py_file`.

Reply:
Wouldn't it actually make more sense to drop the class method altogether, and have the constructor take both types of argument? Seems like it'd actually make the whole structure more robust and usable.

Reply:
Class methods are pretty much the standard way of making alternative constructors in Python. Function overloading in Python is not really the cleanest thing, so I'd rather avoid it if we can.

Reply:
Fixed & squashed